src/EnergyPlus/OutsideEnergySources.cc

```diff
@@ -86,7 +86,7 @@ namespace EnergyPlus::OutsideEnergySources {
 // PURPOSE OF THIS MODULE:
 // Module containing the routines dealing with the OutsideEnergySources
 
-PlantComponent *OutsideEnergySourceSpecs::factory(EnergyPlusData &state, int objectType, std::string objectName)
+PlantComponent *OutsideEnergySourceSpecs::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string objectName)
```

Comment:
Do not pass `std::string` by value. Either pass it by `const &` or better yet use `std::string_view` here.

Reply:
TODO: Check the other factory functions.